tests/other/test_split_module.post.js

```diff
@@ -1,12 +1,25 @@
 function saveProfileData() {
-  // Avoid offset 0 so that we don't upset the stack cookie monster.
-  var offset = 1024;
   var __write_profile = Module['asm']['__write_profile'];
   if (__write_profile) {
-    var len = Module['asm']['__write_profile'](offset, -1);
+    var capacity = 1;
+    var offset;
+    var len;
+    while (true) {
+      offset = _malloc(capacity);
```

Comment:
Why use malloc here?     Why not just write from 1024 all the way to the end of memory?

Reply:
The test was previously doing that and working fine, but when I moved the printing function to after the profile dumping function, the printing stopped working! I assume the static strings I was trying to print were being overwritten by the profile dumper. To fix that I had to properly allocate a buffer, and at that point I thought I might as well test that the whole retrying logic worked correctly since this is essentially what we expect end users to do. I can simplify it by getting rid of the loop if you want, though.